llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -1144,13 +1153,10 @@ bool AccessAnalysis::createCheckForAccess(RuntimePointerChecking &RtCheck,
 
     // When we run after a failing dependency check we have to make sure
     // we don't have wrapping pointers.
-    if (ShouldCheckWrap) {
-      // Skip wrap checking when translating pointers.
-      if (TranslatedPtrs.size() > 1)
-        return false;
-
-      if (!isNoWrap(PSE, AR, Ptr, AccessTy, TheLoop, Assume))
-        return false;
+    if (ShouldCheckWrap &&
+        !isNoWrap(PSE, AR, TranslatedPtrs.size() == 1 ? Ptr : nullptr, AccessTy,
```

Comment:
A direct translation would be
```suggestion
        !isNoWrap(PSE, AR, TranslatedPtrs.size() >= 1 ? Ptr : nullptr, AccessTy,
```
why *exactly* 1?

Reply:
Is was specifically thinking about the case `TranslatedPtrs.empty()`. Before the patch, `isNoWrap` would have been skipped. Not it is executed with `Ptr==nullptr`.